app/Knight.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+
+class Knight:
+
+    knights = []
+
+    def __init__(self, knight: dict) -> None:
+
+        knight["protection"] = 0
+        for person in knight["armour"]:
+            knight["protection"] += person["protection"]
+
+        knight["power"] += knight["weapon"]["power"]
+
+        if knight["potion"] is not None:
+            stats = knight["potion"]["effect"]
+            if "power" in stats:
+                knight["power"] += stats["power"]
+            if "protection" in stats:
+                knight["protection"]\
+                    += stats["protection"]
+            if "hp" in stats:
+                knight["hp"] += stats["hp"]

```

Comment:
You can use ` for` loop for this.
Also the similar recommendation in the checklist:

![image](https://user-images.githubusercontent.com/104218550/202273733-ded1e22a-8552-4dda-b277-f1798b4f2be8.png)

Reply:
Done